src/Bicep.Core/Analyzers/Linter/Rules/UseParentPropertyRule.cs

```diff
@@ -0,0 +1,174 @@
+
+            foreach (var resource in allResources)
+            {
+                if (resource.TypeReference.TypeSegments.Length < 2 ||
+                    resource.Parent is not null ||
+                    resource.TryGetNameSyntax() is not StringSyntax childName)
+                {
+                    continue;
+                }
+
+                var parentType = resource.TypeReference.TypeSegments
+                    .Take(resource.TypeReference.TypeSegments.Length - 1)
+                    .ConcatStrings("/");
+
+                foreach (var parentResource in typeLookup[parentType])
+                {
+                    if (!parentResource.Symbol.IsCollection &&
+                        parentResource.TryGetNameSyntax() is {} parentNameSyntax &&
+                        TryGetParentName(childName, parentNameSyntax) is {} replacement &&
+                        TryCreateDiagnostic(diagnosticLevel, parentResource, resource, replacement) is {} diagnostic)
+                    {
+                        yield return diagnostic;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<string> RemoveFirstLeadingSlash(IEnumerable<string> segments)
+            => segments.Select((x, i) => i == 0 ? x.Substring(1) : x);
+
+        private SyntaxBase? TryGetParentName(StringSyntax parentName, StringSyntax childName)
+        {
+            /* Handle cases such as:
+             * parent: 'abc', child: 'abc/def'
+             * parent: '${abc}', child: '${abc}/${def}'
+             * parent: '${abc}', child: '${abc}/def'
+             * parent: '${abc}-foo', child: '${abc}-foo/bar'
+            */
+
+            if (childName.SegmentValues.Length < parentName.SegmentValues.Length ||
+                childName.Expressions.Length < parentName.Expressions.Length)
+            {
+                // child name must have more or equal string segments and expressions
+                return null;
+            }
+
+            for (var i = 0; i < parentName.SegmentValues.Length - 1; i++)
+            {
+                if (childName.SegmentValues[i] != parentName.SegmentValues[i])
+                {
+                    // apart from the last child segment, all segments must match those of the parent
+                    return null;
+                }
+            }
+
+            var finalParentSegment = parentName.SegmentValues[parentName.SegmentValues.Length - 1];
+            var finalChildSegment = childName.SegmentValues[parentName.SegmentValues.Length - 1];
+
+            if (finalChildSegment.Length <= finalParentSegment.Length ||
+                finalChildSegment[finalParentSegment.Length] != '/' ||
+                finalChildSegment.IndexOf(finalParentSegment) != 0)
+            {
+                // the last common segment must either match exactly, or be equal to <parent>/<something>
+                return null;
+            }
```

Comment:
[nit]: Aren't `finalParentSegment` and `finalChildSegment` already checked for exact equality in the `for` loop above?

Reply:
Not quite - this logic is hard to follow, so I'll try to comment it better. It's possible we have something like:

parent: `foo${abc}bar`, child: `foo${abc}bar/baz`

You'll notice the for loop deliberately checks everything *apart* from the last common string segment (`bar` & `bar/baz` respectively) for equality; whereas we do a substring check for the last common string segment - to handle this case.